lib/connextclient/ConnextClient.ts

```diff
@@ -556,11 +558,22 @@ class ConnextClient extends SwapClient {
     };
   }
 
-  private getBalance = async (currency: string) => {
-    const tokenAddress = this.getTokenAddress(currency);
-    const res = await this.sendRequest(`/balance/${tokenAddress}`, 'GET');
-    const balance = await parseResponseBody<ConnextBalanceResponse>(res);
-    return balance;
+  private getBalance = (currency: string): Promise<ConnextBalanceResponse> => {
+    // check if we already have a balance request that we are waiting a response for
+    // it's not helpful to have simultaneous requests for the current balance, as they
+    // should return the same info.
+    let getBalancePromise = this.getBalancePromises.get(currency);
+    if (!getBalancePromise) {
+      // if not make a new balance request and store the promise that's waiting for a response
+      const tokenAddress = this.getTokenAddress(currency);
+      getBalancePromise = this.sendRequest(`/balance/${tokenAddress}`, 'GET').then((res) => {
+        this.getBalancePromises.delete(currency); // clear the stored promise
```

Comment:
At the moment the pending promise is only cleaned up in case of a happy flow. This cleanup should probably go into `finally` block.